BuildScript.Nuke/GenerateTestMatrix/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,226 @@
+  {
+    var testAssemblyFullPath = testProjectMetadata.AssemblyPaths.Single();
+    var testConfigurationsMetadataValue = testProjectMetadata.TestConfiguration;
+
+    if (string.IsNullOrEmpty(testConfigurationsMetadataValue))
+      throw new InvalidOperationException($"Could not find item metadata 'TestConfiguration' for '{testAssemblyFullPath}'.");
+
+    var rawTestConfigurations = testConfigurationsMetadataValue
+        .Split(';')
+        .Select(x => x.Trim())
+        .Where(x => x != "")
+        .ToArray();
+
+    var unfilteredTestConfigurations = testConfigurationsFactory.CreateTestConfigurations(testProjectMetadata, rawTestConfigurations);
+    var tmpUnfilteredTestConfigurations = unfilteredTestConfigurations.Where(testConfig =>
+        testConfig.ConfigurationID == testProjectMetadata.Configuration.ToLower() &&
+        testConfig.IsNetCoreExecutable ^ testConfig.IsNetFramework
+    ).ToArray();
+    var filteredTestConfigurations = testConfigurationsFilter.GetFilteredTestConfigurations(
+        tmpUnfilteredTestConfigurations);
+
+    return filteredTestConfigurations;
+  }
+
+  private IReadOnlyCollection<TestProjectMetadata> PrepareTestProjectMetadataInput (IReadOnlyCollection<TestProjectMetadata> projectMetadataList)
+  {
+    return projectMetadataList.SelectMany(
+        projectMetadata =>
+            projectMetadata.TargetFrameworks.Select(
+                targetFramework => CreateTestProjectMetadataByTargetFramework(projectMetadata, targetFramework))
+    ).ToList();
+  }
+
+  private TestProjectMetadata CreateTestProjectMetadataByTargetFramework (TestProjectMetadata projectMetadata, string targetFramework)
+  {
+    var projectMetadataTmp = new TestProjectMetadata()
+                             {
+                                 Configuration = projectMetadata.Configuration,
+                                 ProjectPath = projectMetadata.ProjectPath,
+                                 ToolsVersion = projectMetadata.ToolsVersion,
+                                 IsMultiTargetFramework = projectMetadata.IsMultiTargetFramework,
+                                 IsSdkProject = projectMetadata.IsSdkProject,
+                                 AssemblyPaths = new List<string>
+                                                 {
+                                                     projectMetadata.AssemblyPaths.Single(
+                                                         x =>
+                                                             !projectMetadata.IsMultiTargetFramework || x.Contains(targetFramework))
+                                                 },
+                                 TestConfiguration = projectMetadata.TestConfiguration,
+                                 TestSetupBuildFile = projectMetadata.TestSetupBuildFile,
+                                 TargetFrameworks = new List<string> { targetFramework }
+                             };
+    return projectMetadataTmp;
```

Comment:
You could just return immediately, this would lead to the removal of the variable name 🙂